jdecalog commented on 29 Jun 2017
My operating system is (include version):
Debian 8.6
I installed Certbot with (certbot-auto, OS package manager, pip, etc):
Certbot is installed via Debian backport and is up to date (certbot 0.10.2)
I ran this command and it produced this output:
certbot renew failed because the port 443 is already in use (haproxy)
The trace is :
Processing /etc/letsencrypt/renewal/mycert.conf
-------------------------------------------------------------------------------
Cert is due for renewal, auto-renewing...
Starting new HTTPS connection (1): acme-v01.api.letsencrypt.org
Renewing an existing certificate
Performing the following challenges:
tls-sni-01 challenge for mycert
-------------------------------------------------------------------------------
The program haproxy (process ID 20526) is already listening on TCP port 443.
This will prevent us from binding to that port. Please stop the haproxy program
temporarily and then try again. For automated renewal, you may want to use a
script that stops and starts your webserver. You can find an example at
https://certbot.eff.org/docs/using.html#renewal . Alternatively you can use the
webroot plugin to renew without needing to stop and start your webserver.
-------------------------------------------------------------------------------
Cleaning up challenges
Certbot's behavior differed from what I expected because:
The settings in the renewal config file are :
# Options used in the renewal process
[renewalparams]
authenticator = standalone
installer = None
account = xxxxxxxxxx
http01_port = 54321
If is run the following :
certbot certonly --text -d mycert --standalone --preferred-challenges http-01 --http-01-port 54321
it works fine (cert has been generated with this command)